


<!DOCTYPE html>
<html>

<head>
<link rel="stylesheet" href="http://yui.yahooapis.com/pure/0.6.0/pure-min.css">
<link rel="stylesheet" href="../css/layouts/side-menu.css">
<link rel="stylesheet" href="../css/code.css">


<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
});
</script>


<meta name="viewport" content="width=device-width, initial-scale=1">
</head>

<body>

<div id="layout">
    
    <a href="#menu" id="menuLink" class="menu-link">
        
        <span></span>
    </a>

    <div id="menu">
        <div class="pure-menu">
            <a class="pure-menu-heading" href="..">Ontologizer</a>

            <ul class="pure-menu-list">
                
                    
                    <li class="pure-menu-item"><a href="../intro/" class="pure-menu-link">Introduction</a></li>
                    
                
                    
                    <li class="pure-menu-item"><a href="../webstart/" class="pure-menu-link">Webstart</a></li>
                    
                
                    
                    <li class="pure-menu-item"><a href="../invoke-manually/" class="pure-menu-link">Invoke Manually</a></li>
                    
                
                    
                    <li class="pure-menu-item"><a href="../webontologizer/" class="pure-menu-link">Web Ontologizer</a></li>
                    
                
                    
                    <li class="pure-menu-item"><a href="../commandline/" class="pure-menu-link">Command Line</a></li>
                    
                
                    
                    <li class="pure-menu-item"><a href="../input/" class="pure-menu-link">Input</a></li>
                    
                
                    
                    <li class="pure-menu-item"><a href="../howto/" class="pure-menu-link">How To</a></li>
                    
                
                    
                    <li class="pure-menu-item"><a href="../contributing/" class="pure-menu-link">Contributing</a></li>
                    
                
                    
                    <li class="pure-menu-item"><a href="../impressum/" class="pure-menu-link">Impressum</a></li>
                    
                
                    
                


            </ul>
        </div>
    </div>


<div id="main">

    <div class="header">
        <h1>Introduction</h1>
        
    </div>

    <div class="content">
        

<p>A number of methods have been developed to analyze microarray and other high-throughput data derived from measurements of hundreds or thousands of genes or proteins. Depending on the method, the results of analysis consist of a list of many differentially regulated genes or a partitioning of many or all of the genes in the data set into clusters (groups) that putatively share some functionally relevant characteristic. Among the many uses to which the Gene Ontology (GO) has been put is to provide a summary of such lists of genes/proteins according to functional role, biochemical reaction, or location of the gene product in the cell.
The first version of the Ontologizer has been completely redesigned to provide a versatile WebStart or desktop application for the GO term enrichment analysis whose user interface utilizes <a href="http://www.eclipse.org/">Eclipse&rsquo;s</a> Standard Widget Toolkit. It supports not only the standard approach to GO term enrichment analysis but also our parent-child method as described by Grossmann, topology based methods as described by Alexa and our new model-based approach</p>

<p>Following analysis with one of the above methods and optionally a multiple-testing correction procedure, the Ontologizer rows of terms together with their p-values or marginal probabilities, annotation counts and other information. Relevance of a term is indicated by color coding according to the sub ontology to which the term belongs, whereby the intensity of the color correlates with the maginitude of relevance.</p>

<p>Users can click on any term in the table to display properties and results related to the term such as its parents and children, its description, and a list of all genes annotated to the term in the study set. This information is presented as a hypertext in the lower panel with links to parent and child terms. The Ontologizer also provides a tightly integrated graphical display of the results using GraphViz.</p>

<p>The following pages present a brief introduction to the Gene Ontology, the standard &ldquo;term-for-term&rdquo; approach to statistical analysis, and the the parent child approach.</p>

<h2 id="background">Background</h2>

<p>This section intends to give readers a very quick overview of the GO. The <a href="http://www.geneontology.org/">GO homepage</a> is a great source of documentation. Readers are also refered to the original paper on GO by Ashburner and coworkers as well as a <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16381878">recent overview</a> for more information.</p>

<p>The Gene Ontology (GO) has provided a dynamic, controlled vocabulary for describing gene products in any organism. GO contains three extensive subontologies describing molecular function (the biochemical activity of a gene product), biological process (the objective or biological goal to which a gene product contributes) and cellular component (the place in the cell in which the biological activity of a gene product is exerted).</p>

<p>GO contains well over over 20,000 terms, each of which has an accession number, a name, a more detailed definition, and other information relating a term to its parent terms. Individual terms are organized as a directed acyclic graph, whereby the terms form the nodes in the ontology and the arcs the relationships. More specific terms are lower in the graph and terms are related to their parent terms by <em>is-a</em> relationships (e.g. condensed chromosome <em>is-a</em> chromosome) or <em>part-of</em> relationships (e.g. nucleolus is <em>part-of</em> nucleus). In contrast to simpler hierarchical structures, one node in a directed acyclic graph may have multiple parents. For instance, the term <em>DNA replication</em> (GO:0006260) is a child of the term <em>DNA replication and cell cycle</em> (GO:0000067) and also of the term <em>S phase of mitotic cell cycle</em> (GO:0000084). This allows for a more flexible, expressive and detailed description of biological functions.</p>

<p>The GO terms do not themselves describe specific genes or gene products. Rather, collaborating databases generate gene association files consisting of links between genes or gene products and GO terms. Genes and gene products are annotated at the most specific level possible, but are considered to share the attributes of all the parent nodes. Association files have been made available for m organisms, including human, mouse, yeast and Caenorhabditis elegans.</p>

<h2 id="overrepresentation">Overrepresentation</h2>

<h3 id="introduction">Introduction</h3>

<p>We use the designation &ldquo;Term-for-Term (TFT) analysis&rdquo; to refer to the standard method of performing statistical analysis for overrepresentation in GO. Please see the section on Parent-Child analysis for an explanation of this name.</p>

<p>TFT analysis is based on the following assumptions. We are generally interesting in knowing if a subset of genes from a larger population has any special characteristics. We will define the subset as our study group. This could be a set of overexpressed genes from a microarray experiment. Note that there may be several study groups, for instance, overexpressed genes and underexpressed genes. The population group would then be defined as all genes represented on the microarray. We can then ask if the frequency of a annotation to a Gene Ontology term is different for the study group compared to the overall population.</p>

<h3 id="modified-fisher-s-exact-test">Modified Fisher&rsquo;s Exact Test</h3>

<p>Thus, we can imagine a binary random variable for each gene in the population with two mutual exclusive values: 1) The gene is annotated to the GO term in question or 2) The gene is not annotated to the GO term.</p>

<p>Assume the population of all genes contains M genes annotated to some term t and N genes not annotated to the term. The population thuis contains a total of M+N genes. The study set contains k genes, r of which are annotated to the term t. The hypergeometric distribution then gives the probability of sampling exactly r genes annotated term t:</p>

<p><code>$$h_{k,N,M}(r)=\frac{\binom{M}{r}\binom{N}{k-r}}{\binom{N+M}{k}}$$</code></p>

<p>Since we are interested in knowing the probability that a certain number of annotations to a certain GO term occured by chance (with low probabilities corresponding as usual to a high statistical significance), we actually need to calculate the chance of seeing r or more annotations in our study set:</p>

<p><code>$$\sum_r^k h_{k,N,M}(r)$$</code></p>

<p>This is essentially the same as the Fisher Exact Test, except that we are only testing for overrepresentation. The Fisher Exact test, in contrast, tests for either extreme (over- or underrepresentation). It has become the convention in GO analysis not to test for underrepresentation of GO terms, because the biological significance of underrepresentation is much less clear than the significance of overrepresentation.</p>

<h3 id="parent-child-analysis">Parent-Child Analysis</h3>

<p>The Parent-Child method represents a new algorithm for identifying overrepresented Gene Ontology (GO) annotations in gene sets. While the current methods treats each term independently and hence ignores the structure of the GO hierarchy, our approach takes parent-child relationships into account. Over-representation of a term is measured with respect to the presence of its parental terms in the set. This resolves the problem that the standard approach tends to falsely detect an over-representation of more specific terms below terms known to be over-represented. Our approach comes at no additional computational complexity when compared to the standard approach.</p>

<p>The parent-child method is described in detail in <a href="http://www.ncbi.nlm.nih.gov/pubmed/17848398?dopt=Abstract">Grossmann et al. (2007), Improved Detection of Overrepresentation of Gene-Ontology Annotations with Parent-Child Analysis</a> (Available as an Open-Access Bioinformatics Article). A conference report from the RECOMB06 <a href="http://dx.doi.org/10.1007/11732990_9">An Improved Statistic for Detecting Over-Represented Gene Ontology Annotations in Gene Sets</a>, has appeared in Springer&rsquo;s Lecture Notes in Computer Science.</p>

<h3 id="multiple-testing">Multiple Testing</h3>

<p>Since in general we will test up to thousands of GO terms for overrepresentation, we need to apply some correction for multiple testing. At present, the Ontologizer uses a classic Bonferroni correction, meaning it multiplies the nominal p-values calculated as described above by the number of tests performed. This is a very conservative form of correction for multiple testing. We can limit the number of tests performed by deciding not to test GO terms that do not annotate any genes in the population (since the study group is drawn from the population, if no genes are annotated than obviously overrepresentation of the term is impossible). Additionally, if a term annotates only one gene in the population than it is apparent that testing for overrepresentation in the study group has little meaning. Note that the number of genes in the study group annotated to the term does (and should) not need to be taken into account here.</p>

<p>It is possible to perform analysis on any number of groups (clusters) of genes simultaneously. The Ontologizer does not perform multiple testing correction based on the number of clusters analysed. Depending on the question posed by the user, it may or may not to be appropriate to do so.</p>

<p>Finally, note that Gene Ontology annotations are made to the most specific term possible. All ancestors of the term are considered to be implicitly annotated. Therefore, if we are calculating the total annotations of a term, we need to count annotations to all (more specific) descendents of the term also. Note that we need to avoid introducing &ldquo;extra&rdquo; (duplicate) counts if there are multiple paths from a descendent term to an ancestor term, or if two distinct descendents of a term are annotated for a certain gene. A discussion of these issues is available in Robinson et al. (2004) <a href="http://www.ncbi.nlm.nih.gov/pubmed/15254011?dopt=Abstract">Gene-Ontology analysis reveals association of tissue-specific 5&rsquo; CpG-island genes with development and embryogenesis</a></p>

<h3 id="potential-problems-with-the-term-browser-and-help-window-of-the-ontologizer">Potential Problems with the Term Browser and Help Window of the Ontologizer</h3>

<p>With some combinations of debian linux and firefox, we have noticed a problem with the Ontologizer finding the libraries needed to allow browsing of GO terms and the show the HTML format of the help system. If you notice this problem, it may help to set the variable MOZILLA_FIVE_HOME in the file /etc/environment as follows:</p>

<pre><code>MOZILLA_FIVE_HOME=/usr/lib/xulrunner and to install the latest version of the packages &quot;xulrunner&quot; and &quot;libxul-dev&quot; with apt-get or aptitude.
</code></pre>

<p>The Ontologizer should start properly with all common Windows browsers (we have tested it with IE6, IE7 and Firefox).</p>

<h3 id="macintosh">MacIntosh</h3>

<p>The Ontologizer is built using GUI elements from the Standard Widget Toolkit (SWT) known from Eclipse. SWT has been built and test for Windows and Linux systems and the Ontologizer should work without problems on machines running Linux or modern versions of Windows. Unfortunately, the SWT for Mac OSX is not yet advanced enough, and therefore we decided to declare Ontologizer for Macintosh as unsupported at this time. Embarassed</p>

    </div>
</div>

</div>
</div>
</div>

<script src="../js/ui.js"></script>

</body>
</html>

